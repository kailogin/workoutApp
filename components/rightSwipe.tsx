import React from "react";
import { TouchableOpacity } from "react-native";
import { MaterialIcons } from "@expo/vector-icons";

import { Colors } from "../utils/theme";

interface RightSwipeProps {
  handleClick: () => void;
}

export const RightSwipe = ({ handleClick }: RightSwipeProps) => (
  <TouchableOpacity
    onPress={handleClick}
    style={{
      alignItems: "center",
      backgroundColor: Colors.RED,
      borderTopEndRadius: 12,
      borderBottomEndRadius: 12,
      flex: 0.5,
      justifyContent: "center",
      marginBottom: 6,
    }}
  >
    <MaterialIcons name="delete" size={24} color={Colors.WHITE} />
  </TouchableOpacity>
);
