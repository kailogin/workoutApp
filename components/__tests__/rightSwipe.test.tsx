import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { RightSwipe } from "../rightSwipe";

const TEXT = "Test Text";

test(`BaseText snapshot`, () => {
  const snap = renderer
    .create(<RightSwipe handleClick={() => null} />)
    .toJSON();

  expect(snap).toMatchSnapshot();
});
