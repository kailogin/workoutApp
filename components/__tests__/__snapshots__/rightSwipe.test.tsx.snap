// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F76A6A",
      "borderBottomEndRadius": 12,
      "borderTopEndRadius": 12,
      "flex": 0.5,
      "justifyContent": "center",
      "marginBottom": 6,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Montserrat",
        },
        Object {
          "color": "#fff",
          "fontSize": 12,
          "fontWeight": "600",
          "padding": 8,
        },
      ]
    }
  >
    DELETE
  </Text>
</View>
`;
